catalyst-gateway/tests/Earthfile

```diff
@@ -1,26 +1,89 @@
 VERSION 0.7
 
+build:
+    FROM ../+builder
+    RUN cargo build -p cat-gateway --release
+    SAVE ARTIFACT target/release/cat-gateway cat-gateway
+
+cat-gateway-publish:
```

Comment:
For cat-ci to run this properly, it should be a `package` target.
```
package-cat-gateway:
```

Also, ideally, this would be the actual package we would be deploying if all tests pass, so it should probably be in the main Earthfile, and just referenced from there.

Reply:
Sure I'll move it to the main Earthfile